BazelExtensions/extensions.bzl

```diff
@@ -83,79 +83,107 @@ def acknowledged_target(name,
         visibility=["//visibility:public"]
     )
 
-def _make_module_map(pod_name, module_name, deps, is_system):
-    # Up some dirs to the compilation root
-    # bazel-out/ios_x86_64-fastbuild/genfiles/external/__Pod__
-    relative_path = "../../../../../../"
 
-    system_tag = " [system] "
-    template = "module " + module_name + (system_tag if is_system else "" ) + " {\n"
-    template += "    export * \n"
+def _get_module_map_headers(deps):
+    headers_list = []
     for provider in deps:
         for input_file in provider.files.to_list():
             if input_file.path.endswith(".hmap"):
                 continue
-            hdr = input_file
-            template += "    header \"%s%s\"\n" % (relative_path, hdr.path)
-    template += "}\n"
-    return template
-
-def _gen_module_map_impl(ctx):
-  out = _make_module_map(ctx.attr.pod_name, ctx.attr.module_name, ctx.attr.hdrs, ctx.attr.is_system)
-  ctx.actions.write(
-      content=out,
-      output=ctx.outputs.module_map
-  )
-  objc_provider = apple_common.new_objc_provider(
-      module_map=depset([ctx.outputs.module_map])
-  )
-  return struct(
-     files=depset([ctx.outputs.module_map]),
-     providers=[objc_provider],
-     objc=objc_provider,
-     headers=depset([ctx.outputs.module_map]),
-  )
+            if input_file.path.endswith(".hpp"):
+                continue
+            if input_file.path.endswith(".modulemap"):
+                continue
+            headers_list.append(input_file)
+    return headers_list
+
+def _module_map_impl(ctx):
+    module_name = ctx.attr.module_name
+    deps = ctx.attr.hdrs
+    swift_header = ctx.attr.swift_header
+    headers_list = _get_module_map_headers(deps)
+
+    # Up some dirs to the compilation root
+    # bazel-out/ios_x86_64-fastbuild/genfiles/external/__Pod__
+    relative_path = "../../../../../../"
```

Comment:
Yikes! Is this a stable assumption? Is there a variable or macro that we can use to get this absolute path?

Reply:
😂  The headers have to be relative to the build file location which is stable e.g. in `Vendor|external/PodName`. There's currently no code that moves them to another location

Reply:
Right, got that. I mean could be use `$(location)` or `$(rootpath)` or [similar](https://docs.bazel.build/versions/2.0.0/be/make-variables.html) here?

Reply:
Because of caching, we cannot write an absolute path into an artifact. That being said, I've updated the PR to compute the value dynamically. It's useful if we use the rule in another place